code/datums/black_market_sellable.dm

```diff
@@ -69,78 +71,361 @@ var/list/black_market_sellables = list()
 		. = 999
 	else
 		. = round_demand
+		
+/datum/black_market_sellable/proc/reduce_demand()
+	if(round_demand != -1 || round_demand != 0)
+		round_demand--
 
+/datum/black_market_sellable/proc/on_setup()
+	return
+		
 /datum/black_market_sellable/proc/purchase_check(var/obj/input, var/mob/user) //Returns "VALID" if the purchase is valid, otherwise will give an error message with what is returned.
 	return VALID
 	
 /datum/black_market_sellable/proc/determine_payout(var/obj/input, var/mob/user, var/payout) //Override for extra price calculations. Be sure to cast the given var/obj/ into the proper type.
 	return get_price()
-
-/datum/black_market_sellable/proc/after_sell(var/obj/input, var/mob/user) 
-	return
-	
-	
 	
+/datum/black_market_sellable/proc/after_sell(var/obj/input, var/mob/user)
+	return
+
 	
+
+/datum/black_market_sellable/weapons
+	category = "Firearms and War Implements"
 	
+/datum/black_market_sellable/weapons/ion_rifle
+	name = "Ion Rifle"
+	item = /obj/item/weapon/gun/energy/ionrifle
+	no_children = 1
+	demand_min = 1
+	demand_max = 3
+	price_min = 100
+	price_max = 200
+	sps_chance = 10
+	display_chance = 40
 	
+/datum/black_market_sellable/weapons/vector
+	name = "Kriss Vector"
+	item = /obj/item/weapon/gun/projectile/automatic/vector
+	no_children = 1
+	desc = "The submachine gun, silly."
+	demand_min = 1
+	demand_max = 3
+	price_min = 100
+	price_max = 200
+	sps_chance = 10
+	display_chance = 40
+
+/datum/black_market_sellable/weapons/railgun
+	name = "Railgun"
+	item = /obj/item/weapon/gun/projectile/railgun
+	no_children = 1
+	desc = "We'll pay triple for a railgun loaded with a capacitor of 1 GW charge or greater."
+	demand_min = 1
+	demand_max = 1
+	price_min = 200
+	price_max = 300
+	sps_chance = 15
+	display_chance = 100
 	
+/datum/black_market_sellable/weapons/railgun/determine_payout(var/obj/input, var/mob/user, var/payout)
+	if(istype(input,/obj/item/weapon/gun/projectile/railgun))
+		var/obj/item/weapon/gun/projectile/railgun/gun = input
+		if(gun.loadedcapacitor && gun.loadedcapacitor.stored_charge/5000000 >= 200) //1 GW
+			return payout*3
+	return payout
 	
+/datum/black_market_sellable/weapons/transfer_valve
+	name = "Tank Transfer Valve"
+	item = /obj/item/device/transfer_valve/
+	no_children = 0
+	desc = "Looking for NT-brand explosive valves. The tanks attached are irrelevant, send it empty if you want."
+	demand_min = 1
+	demand_max = 2
+	price_min = 400
+	price_max = 500
+	sps_chance = 70
+	display_chance = 50
+
+/datum/black_market_sellable/weapons/ied
+	name = "Improvised Explosive Device"
```

Comment:
50 - 120 bucks for an item you can make with a $10 soda can and stuff you find in the tool storage is a bit much

Reply:
Changed it to 20-30, more reasonable?